modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -703,65 +776,23 @@ private CompletableFuture<Void> scaleDownAwaiting(int zoneId, boolean immediateS
     }
 
     /**
-     * Returns the future with data nodes of the specified zone.
+     * Returns the data nodes of the specified zone.
      *
      * @param zoneId Zone id.
      * @return Future.
      */
-    private CompletableFuture<Set<String>> getDataNodesFuture(int zoneId) {
+    private Set<String> dataNodes(int zoneId) {
```

Comment:
Please also consider removing `getDataNodesByZoneId` in that case.

Reply:
removed